feat: add expo-notifications dependency and implement QR code notific…

anaelchevillard opened the pull request:
…ation system

- Added expo-notifications to package.json for handling notifications.
- Created a new utility for managing QR code notifications related to meal bookings.
- Implemented functions to check lunch breaks, get lunch break start times, and verify meal bookings.
- Developed notification scheduling for QR codes, including daily notifications during lunch breaks.
- Added functionality to handle notification responses and navigate to QR code display.
- Included permission requests for notifications and a test function for testing QR code notifications.

![Contribution](https://github.com/PapillonApp/papillon-v8/raw/main/.github/assets/contribution_header.png)

# Règles de contribution
> [!CAUTION]
> Afin de garantir une application stable et pérenne dans le temps, nous t'invitons à vérifier que tu as bien respecté les règles de contribution. Sans cela, ta Pull Request ne pourra pas être examinée.

- [x] Cette Pull Request porte sur une seule fonctionnalité ou un seul correctif.
- [x] Cette Pull Request n'est pas faite essentiellement avec de l'IA.
- [x] Pour tout changement majeur, j’ai créé une issue afin d’échanger avec les mainteneurs de Papillon sur la meilleure façon de l’intégrer.
- [x] Ma Pull Request respecte les conventions Conventional Commits et Conventional Branch ainsi que les conventions de codage de l'application.
- [x] J’ai testé mes modifications sur iOS et Android, et l’application fonctionne correctement.
- [x] J’emploie un langage informel, clair et concis dans mes messages.
- [x] J’ai documenté mes changements de manière appropriée, soit dans la description de la Pull Request, soit dans le GitBook.
- [x] J’ai ajouté les traductions nécessaires dans au moins un fichier de langue.

# Résumé des changements

> [!NOTE]
> Ajout d'une notification lors de la pause méridienne pour accéder rapidement à au QR Code du service de restauration. Avec le fichier qrcode-notification.ts Ajout de la lib Expo Notification. Ajout de mode de test dans le DevMode.

> [!NOTE]
>  Normalement, Papillon regarde si on a un système compatible QR Code (Izly ou Turboself), regarde si on a réservé le jour qu'on est. Puis, essaye de trouver le début de la pause méridienne et quand elle arrive, envoie la notification. De base, je voulais faire une Live Activities, mais j'ai eu les souvenirs de Lucas galérant avec le Swift, je suis alors passé à la notification.

> [!CAUTION]
> Cela fait LONGTEMPS que je n'ai pas autant codé en React et TS. De plus, la structure de papillon a changé (V7 to V8). Ainsi, il doit y avoir beaucoup d'erreurs, quelques bugs et un peu de code spaghetti (rien de sûr, mais je préfère prévenir). J'ai galéré avec les API Izly et Turboself. Je n'ai pas de compte Turboself ou Izly et je suis en vacances, ce qui fait que je n'ai pas pu tester l'entièreté de l'ajout.

# Capture(s) d'écran
<img width="764" height="320" alt="CleanShot 2025-10-22 at 20 44 27@2x" src="https://github.com/user-attachments/assets/7a500cbd-43ea-4c40-a61f-d282d6e519c3" />

Le bouton "Test notification QR Code", envoie une simple notification et génère un faux QR Code
Le bouton "Test notification intelligente" fait tout le processus, de vérif de compte etc. C'est plus pour vous, car je n'ai pas pu tester (raison expliquée au dessus).

# Informations supplémentaires
> [!NOTE]
> C'était très difficile et je ne sais pas si ça fonctionne !

godetremy commented:
Hello,

Cette PR à déjà était soumise. La republier sans modification apparente n'as pas d'intérêt. Je la close.

Merci 🙃

codeuriii commented:
mais tu regarde les commits non ? tu vois bien que y a des modifs nan ?
faut regarder avant de juger

godetremy commented:
> mais tu regarde les commits non ? tu vois bien que y a des modifs nan ? faut regarder avant de juger

L'intégration reste la même. C'est celle-ci que nous refusons.

codeuriii commented:
j'ai pas compris

anaelchevillard committed:
feat: add expo-notifications dependency and implement QR code notification system

- Added expo-notifications to package.json for handling notifications.
- Created a new utility for managing QR code notifications related to meal bookings.
- Implemented functions to check lunch breaks, get lunch break start times, and verify meal bookings.
- Developed notification scheduling for QR codes, including daily notifications during lunch breaks.
- Added functionality to handle notification responses and navigate to QR code display.
- Included permission requests for notifications and a test function for testing QR code notifications.

anaelchevillard commented:
J'ai aussi rencontré des bugs que je n'arrive pas à régler, sûrement par manque de connaissance. Mais à certains moments, quand je fais Test notification QR Code, la page du QR Code s'affiche plusieurs fois. Ce n'est pas h24 et je n'arrive pas à régler le bug.

anaelchevillard committed:
Merge branch 'main' into eatcard-notification
